src/read.rs

```diff
@@ -66,3 +69,21 @@ where
         Ok(data)
     }
 }
+
+/// Loop while the given function returns true or the timeout is reached.
+fn while_with_timeout<E, D, F>(delay: &mut D, func: F, timeout_us: u16) -> Result<(), DhtError<E>> 
+where
+    D: DelayUs<u8>,
+    F: Fn() -> Result<bool, E>
+{
+    let mut count = 0;
+
+    while func()? {
+        delay.delay_us(10_u8);
+        count += 1;
+        if count >= timeout_us {
+            return Err(DhtError::Timeout);
+        }
+    }
+    Ok(())
+}
```

Comment:
I switched the polarity of the `wait` function, I think it reads better (nothing to do with this PR really), WDYT?

```suggestion
/// Wait until the given function returns true or the timeout is reached.
fn wait_until_timeout<E, D, F>(delay: &mut D, func: F, timeout_us: u8) -> Result<(), DhtError<E>>
where
    D: DelayUs<u8>,
    F: Fn() -> Result<bool, E>,
{
    for _ in 0..timeout_us {
        if func()? {
            return Ok(());
        }
        delay.delay_us(1_u8);
    }
    Err(DhtError::Timeout)
}
```